### ISSUE TITLE
Get rid of Bootstrap style imports

### ISSUE THREAD

**Issue by valerybugakov** [sourcegraph/sourcegraph#29200]
## Problem statement

We depend on Bootstrap styles in our global styles a lot. But we don't plan to move forward with Bootstrap because we're working on our design system called Wildcard. To ease the transition to our styles and gain more control over visual changes, we want to move Bootstrap styles _that we need_ into our codebase and eventually drop Bootstrap dependency.

## Success criteria

All Bootstrap CSS rules that we need are incorporated in our codebase. 

## Linked issues

- Bootstrap removal
  - #29252
  - #29253
  - #29355
  - #29356
  - #30279
  - #30280
  - #30281
  - #31574
  - #32438
  - #32439
  - #32458
  - #32819
  - #32820
  - #37324
  - #37325
  - #37914
  - #40293
  - #40294
  - #40295
  - #40296
- Post removal clean up
  - #41397
  - #41398
  - #41399

## Implementation details

Let's start with these SCSS files and create a separate PR for each one:
1. `client/branded/src/global-styles/list-group.scss`
2. `client/branded/src/global-styles/nav.scss`

Notes:
1. We want to migrate away from SCSS variables to CSS variables when we bring Bootstrap styles to our codebase.
    - Some Bootstrap SCSS variables redefined on our side are located here: `client/branded/src/global-styles/index.scss`. Keep that in mind while migrating to CSS variables. 
2. We don't need to bring all styles defined in the imported Bootstrap package. Check what styles do we use in the codebase first.

## Time estimate

- Pull requests with ~450 lines changed should take **6 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
- Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

/cc @taylorsperry @muratsu

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, let's also remove the following imports and copy CSS rules that we need to our own CSS files:

1. `@import 'bootstrap/scss/progress';`
2. `@import 'bootstrap/scss/button-group';`

**Comment by gitstart-sourcegraph** [comment]
Hi @valerybugakov here are some bootstrap imports on codebase we could work on as part of this issue, kindly take a look.

- [ ]  Button `@import 'bootstrap/scss/buttons'`
- [ ]  Forms `@import 'bootstrap/scss/forms'`
- [ ] Custom Forms `@import 'bootstrap/scss/custom-forms`
- [ ] Input Group `@import 'bootstrap/scss/input-group'`
- [ ] Dropdown `@import 'bootstrap/scss/dropdown'`
- [ ] Popover `@import 'bootstrap/scss/popover'`

**Comment by valerybugakov** [comment]
- [ ]  Button `@import 'bootstrap/scss/buttons'` — https://github.com/sourcegraph/sourcegraph/pull/30096
- [x]  Forms `@import 'bootstrap/scss/forms'`
- [x] Custom Forms `@import 'bootstrap/scss/custom-forms`
- [x] Input Group `@import 'bootstrap/scss/input-group'`
- [ ] Dropdown `@import 'bootstrap/scss/dropdown'`  — https://github.com/sourcegraph/sourcegraph/issues/27695
- [ ] Popover `@import 'bootstrap/scss/popover'` — https://github.com/sourcegraph/sourcegraph/issues/27695

@gitstart-sourcegraph, let's create issues and spec PRs for selected items from the list above.

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, let's also extract `@import 'bootstrap/scss/grid`. The file is pretty small: ETA 2-3h.

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, created [an issue](https://github.com/sourcegraph/sourcegraph/issues/39285) to unblock bootstrap removal in the VSCode package. Let me know if you have questions!

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, after addressing feedback [here](https://github.com/sourcegraph/sourcegraph/pull/39286), could you take a look at importing the remaining Bootstrap imports into our codebase? We want to bring utility classes and mixins used by our codebase. Please drop classes and mixins if they are not used anywhere.

```scss
@import 'bootstrap/scss/functions';
@import 'bootstrap/scss/variables';
@import 'bootstrap/scss/mixins';
@import 'bootstrap/scss/reboot';
@import 'bootstrap/scss/utilities';
```

Let's split this work into multiple PRs – one PR per import removed.
